src/main/java/com/hazelcast/aws/EcsClientStrategy.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.impl.Constants;
+import com.hazelcast.aws.impl.DescribeNetworkInterfacesRequest;
+import com.hazelcast.aws.impl.DescribeTasksRequest;
+import com.hazelcast.aws.impl.Ec2OperationClient;
+import com.hazelcast.aws.impl.EcsOperationClient;
+import com.hazelcast.aws.impl.ListTasksRequest;
+import com.hazelcast.aws.utility.Environment;
+import com.hazelcast.aws.utility.MetadataUtils;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.hazelcast.aws.impl.Constants.AWS_EXECUTION_ENV_VAR_NAME;
+import static com.hazelcast.aws.impl.Constants.EC2_PREFIX;
+import static com.hazelcast.aws.utility.StringUtils.isNotEmpty;
+
+/**
+ * Strategy for discovery of Hazelcast instances running under ECS / Fargate
+ */
+class EcsClientStrategy extends AwsClientStrategy {
```

Comment:
I don't see any unit test testing this class. Is it possible (or would it make sense) to add one?